src/journal/lookup3.c

```diff
@@ -319,7 +319,7 @@ uint32_t jenkins_hashlittle( const void *key, size_t length, uint32_t initval)
      * still catch it and complain.  The masking trick does make the hash
      * noticeably faster for short strings (like English words).
      */
-#if !VALGRIND && !defined(__SANITIZE_ADDRESS__)
+#if !VALGRIND && !defined(__SANITIZE_ADDRESS__) && !HAS_FEATURE_ADDRESS_SANITIZER
```

Comment:
`__SANTIZE_ADDRESS__` is for `gcc` and `HAS_FEATURE_ADDRESS_SANITIZER` is for `clang`. If I remove one of them, either `gcc` or `clang` won't be happy.

Reply:
But it's our own code that defines HAS_FEATURE_ADDRESS_SANITIZER, no? (I mean, you added a patch to this very PR that defines it). Hence why not set that macro too on gcc? I mean, I think it's much nicer to abstract the differences of gcc and llvm away when setting the macro instead of checking both everytime we need them...

Reply:
I see. Yes I agree that makes sense but I would prefer it if it could be done in a separate PR at least after making sure that ClusterFuzz is happy with the shenanigans I'm doing here. I tested it locally with all the corpora, sanitizers, fuzzers and engines but it isn't always enough unfortunately.

Reply:
On the other hand, given that this PR is supposed to handle the difference between `gcc` and `clang` I think it's still useful to have something like `HAS_FEATURE_ADDRESS_SANITIZER`.

Reply:
For example, we shouldn't skip destructors if `systemd` has been built with `gcc`.

Reply:
More generally, I'm wondering why anyone would want to always see these warnings running all the commands where `_DEFINE_MAIN_FUNCTION` is used.